### ISSUE TITLE
simple-cipher: Random key length and symbolic values in canonical data

### ISSUE THREAD

**Issue by cmccandless** [exercism/problem-specifications#1250]
### From the [DESCRIPTION](https://github.com/exercism/problem-specifications/blob/master/exercises/simple-cipher/description.md)
> If someone doesn't submit a key at all, generate a truly random key of at least 100 characters in length.

### From the canonical data
https://github.com/exercism/problem-specifications/blob/267f1eae9bbe5938d015f64b2e22809c4abd7a77/exercises/simple-cipher/canonical-data.json#L21-L28

### The problem
While trying to implement the canonical data for this exercise in the Python track, I ran into the following:
```Bash
self = <simple_cipher_test.RandKeyCipherTest testMethod=test_can_decode>

    def test_can_decode(self):
        cipher = Cipher()
>       self.assertEqual(cipher.decode(cipher.key), 'aaaaaaaaaa')
E       AssertionError: 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa[45 chars]aaaa' != 'aaaaaaaaaa'
E       - aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
E       + aaaaaaaaaa
```

To summarize, a face-value translation of the canonical data results in failure, because in the canonical data no substring length is specified for `cipher.key`. Thus, when then full string is used, the length of the actual value is much greater than the length of the expected value.

**Comment by rpottsoh** [comment]
I am borrowing from csharp here...  How about something like this?
```json
        {
          "description": "Can decode",
          "property": "decode",
          "input": {
            "ciphertext": "cipher.key.Substring(0, 10)"
          },
          "expected": "aaaaaaaaaa"
        },
```

**Comment by ErikSchierboom** [comment]
@cmccandless I've run into the same problem, and as @rpottsoh noted, fixed it by manually doing a substring comparison. We should probably choose one of @rpottsoh's two suggestions and fix the canonical data.

**Comment by cmccandless** [comment]
> We should probably choose one of @rpottsoh's two suggestions and fix the canonical data.

Might I offer a third?

```json
{
          "description": "Can decode",
          "property": "decode",
          "input": {
            "ciphertext": "cipher.key[:10]"
          },
          "expected": "aaaaaaaaaa"
        },
```

I'm not sure if I like this more or less than `cipher.key.substring(0, 10)`, but slice syntax is more concise and common enough among languages that I think it still conveys the correct idea. If we do use slice syntax, we might want to add a note about it, potentially containing a link to a more thorough explanation of slice syntax for those who are unfamiliar.

On that note, I may have just talked myself out of slice syntax due to the amount of additional content that it might add to the canonical data. I'll still put this idea out there anyway for more input.

**Comment by ErikSchierboom** [comment]
Hmm, I'm not sure I like the slicing syntax. While it is concise, to me it is less comprehensible than the other two options.

**Comment by cmccandless** [comment]
> While it is concise, to me it is less comprehensible than the other two options.

Which is why I thought it would warrant additional notes, and why by the end of typing that idea I had personally talked myself out of liking it. I vote for `cipher.key.substring(0, 10)` since it conveys both a starting index and length, while `"ciphertextLength": 10` does not specify a starting index.

**Comment by rpottsoh** [comment]
`cipher.key.substring(0, 10)` :+1:

**Comment by ErikSchierboom** [comment]
I also like `cipher.key.substring(0, 10)`! 👍

**Comment by cmccandless** [comment]
Sounds like we have a consensus on a solution. I will create a PR; technically, I believe this would be a minor version bump, since the output is changing. However, it could also arguably be only a patch bump, since the intent of the output has not changed.

@rpottsoh @ErikSchierboom What's our consensus on minor/patch version bump?

**Comment by ErikSchierboom** [comment]
I think this would qualify as a minor version bump: https://github.com/exercism/problem-specifications/blob/master/README.md#minor-version-changes

The test structure remains the same, but the contents have changed.

**Comment by rpottsoh** [comment]
The input is changing so I would favor the version bumping to: 1.1.0
